src/Router.sol

```diff
@@ -29,7 +32,7 @@ contract Router is IRouter, EIP712, FeeGenerator {
     /// @notice Mapping for recording valid signers
     mapping(address signer => bool valid) public signers;
 
-    /// @notice Transient address for recording `msg.sender` which resets to `_INIT_USER` after each execution
+    /// @notice Transient address for recording `msg.sender` which resets to `_INIT_CURRENT_USER` after each execution
     address public currentUser;
```

Comment:
Considering this variable has different purpose now, how about changing the terms here like:
`currentUser` -> `currentState`
`_PAUSED` -> `_PAUSED`
`_INIT_CURRENT_USER` -> `_READY`
`onlyInitCurrentUser` -> `whenReady`

Reply:
The status change method seems to be good, only a little strange 🤔 :
whenReady modifier change currentState is
`currentState = msg.sender`

Reply:
Considering `currentUser` is a public state that's frequently accessed by external contracts like Agent and Callbacks, using `currentState` could introduce some ambiguity, since it's typically meant to indicate the current user.

An alternative could be changing `currentState` to an internal `_currentState`, and introducing a `getCurrentUser` view function. But this could replace all instances of `currentUser` in this repo.

Reply:
After discussion, keep as-is but check `_PAUSED` in `pause` for consistency and intention of this check, in https://github.com/dinngo/composable-router-contract/pull/77/commits/51871b7390e9eb183cfa7315d41b85da8c41312c.